spec/system/admin/reports_spec.rb

```diff
@@ -139,14 +139,15 @@
       # Unlocking the breakpoint will continue execution of the controller.
       breakpoint.unlock
 
-      # We have to wait to be sure that the "loading" spinner won't appear
-      # within the next half second. The default wait time would wait for
-      # 10 seconds which slows down the spec.
-      using_wait_time 0.5 do
-        page.has_selector? ".loading"
-      end
+      # Now the controller response will show the loading spinner again and
+      # the fallback mechanism will render the report later.
+      expect(page).to have_selector ".loading"
+
+      # Wait for the fallback mechanism:
+      sleep 3
```

Comment:
I can see the need for the sleep here, and a few seconds is probably no big deal. 
But it caused me to think of a way around it: if the wait time was a ruby constant (eg in `ReportsController` or `ReportsHelper`), we could mock it to be a much smaller value for the spec.

Reply:
Yes, we could. And if this happened multiple times then I would probably do it. But a one-time wait is probably worth using the original production code. It's simpler this way.